### PR TITLE
Add stringification to ParsedDocument.

### PR DESCRIPTION
**Pull request by rictic**
Doesn't yet handle inline documents. See: https://github.com/Polymer/polymer-analyzer/issues/315

### CODE REVIEW

File: src/css/css-document.ts [hunk 1]
```diff
@@ -86,4 +87,13 @@ export class ParsedCssDocument extends ParsedDocument<shady.Node, Visitor> {
   sourceRangeForNode(_node: shady.Node): SourceRange {
     throw new Error('Not implemented');
   }
+
+  stringify() {
+    return cssbeautify(
+               shadyStringifier.stringify(this.ast),
```
Comment (justinfagnani) [inline src/css/css-document.ts]: This is my daily venting against clang.

File: src/html/html-document.ts [hunk 2]
```diff
@@ -101,6 +101,22 @@ export class ParsedHtmlDocument extends ParsedDocument<ASTNode, HtmlVisitor> {
       }
     };
   }
+
+  stringify() {
+    // TODO(issues/315): serialize inline documents correctly.
+    let result = parse5.serialize(this.ast);
+
+    // Strip out inferred boilerplate nodes that are injected.
```
Comment (justinfagnani) [inline src/html/html-document.ts]: hmm... they're not always injected. What about checking the original source to see if they are there?

File: src/json/json-document.ts [hunk 3]
```diff
@@ -54,4 +54,8 @@ export class ParsedJsonDocument extends ParsedDocument<Json, Visitor> {
   sourceRangeForNode(_node: Json): SourceRange {
     throw new Error('Not Implemented.');
   }
+
+  stringify() {
+    return JSON.stringify(this.ast, null, 2);
```
Comment (justinfagnani) [inline src/json/json-document.ts]: FYI, I found this JSON parser that can report source positions: https://github.com/dscape/clarinet

Would need some work to use...

File: src/html/html-document.ts (hunk 2 above)
Comment (rictic) [inline src/html/html-document.ts]: I do. L112: `!this.contents.includes(`<${tagName}`)`

Comment (justinfagnani) [inline src/html/html-document.ts]: oops :/

Comment (justinfagnani) [inline src/html/html-document.ts]: alternatively, can you check the `location`?

File: src/json/json-document.ts (hunk 3 above)
Comment (rictic) [inline src/json/json-document.ts]: +1 to this or something similar. Writing a JSON parser is really easy if it comes to that.

Looks like what we'd do here is query the state of the input stream as we got events and either have a WeakMap alongside (if we only care about objects and arrays) or build up our own kind of annotated-node based tree?

Comment (justinfagnani) [inline src/json/json-document.ts]: Yeah, something like that. I didn't look very hard, so there might be a non-SAX parser that does this too. JSON isn't so important in the scheme of things though.

**Pull request merged**